src/harness/reference_models/geo/utils.py

```diff
@@ -372,6 +372,45 @@ def PolygonsAlmostEqual(poly_ref, poly, tol_perc=10):
           < tol_perc/100. * GeometryArea(poly_ref))
 
 
+def ShrinkAndCleanPolygon(poly, shrink_deg, min_poly_ratio=0.01):
+  """Shrinks and cleans a polygon.
+
+  This shrinks a polygon and makes sure small polygons due to shrinking
+  artifacts are removed.
+
+  Args:
+    poly: The polygon to shrink, either as shapely or geojson dict.
+    skrink_deg: The shrink amount in degrees. 1e-2 corresponds approximately
+      to 1km
+    min_poly_ratio: All small polygon smaller than this ratio are removed. These
+      are usually caused by artifacts during the shrinking operation using the
+      shapely `buffer(-shrink_deg)` operation. Use negative value for expansion.
+
+  Returns:
+    The shrinked polygon in same format as the input (ie either shapely or
+    geojson dict).
+
+  Raises:
+    ValueError: if the shrinking cannot produce a single Polygon within the
+      `min_poly_ratio` constraint.
+  """
+  is_json = isinstance(poly, dict)
+  if is_json:
+    poly = ToShapely(poly)
+  if not isinstance(poly, sgeo.Polygon):
+    raise ValueError('Input is not a polygon.')
+  poly = poly.buffer(shrink_deg)
+  if isinstance(poly, sgeo.MultiPolygon):
+    # Cleanup multi-polygon.
+    max_poly_area = max(p.area for p in poly)
+    polys = [p for p in poly if p.area > min_poly_ratio * max_poly_area]
+    if len(polys) > 1:
+      raise ValueError('Impossible to shrink polygon '
+                       'into a proper single polygon')
+    poly = polys[0]
```

Comment:
do we always return the first polygon in multi-polygon case ? or should we just simply raise exception ?

Reply:
After trying to remove spurious polygons with small area, if we still have a MultiPolygon, then we raise an exception and stop: the configuration is bad, as the spec speficies that a PPA shall be a single polygon.
It is the task of the SAS (and of the ITS for the test cases) to feed single Polygon.
In fact  we might make another PR with some validation code, such as if a PPA config contains something else than a single Polygon, it stops with an error to avoid such occurence

Reply:
thanks. I agree that we shall check at the config time.